frontend/src/useStepTime.js

```diff
@@ -11,6 +11,23 @@ function useStepTime() {
     } 
   });
 
+  function updateVisualThreshold(forceData) {
```

Comment:
The changes you made to modify the background color based on force data are a great visual enhancement! The logic works well and adds useful feedback for the user. However, I suggest that we modularize the code for clarity and maintainability. By separating the force data parsing and color-changing logic into their own functions, we can easily modify or extend the functionality down the line. Additionally, this will keep the handle_data_streaming() function focused on its main responsibility of managing the WebSocket connection.